### PR TITLE
R headers

### PR DESCRIPTION
**Pull request by philippe44**
This is not a functional PR, just a decomposition of the 3 steps of the remote-header plugin

### CODE REVIEW

File: Slim/Formats/Movie.pm [hunk 1]
```diff
@@ -200,4 +206,82 @@ sub findFrameBoundaries {
...
+	# put at least 16 bytes after mdat or it confuses audio::scan (and header creation)
+	my $fh = File::Temp->new();
+	$fh->write($args->{_scanbuf} . pack('N', $args->{_audio_size}) . 'mdat' . ' ' x 16);
+	$fh->seek(0, 0);
+
+	my $info = Audio::Scan->scan_fh( mp4 => $fh )->{info};
+	$info->{fh} = $fh;
+	$info->{audio_offset} = $args->{_mdat_} + 8;
+	
+	# MPEG-4 audio = 64,  MPEG-4 ADTS main = 102, MPEG-4 ADTS Low Complexity = 103
```
Comment (philippe44) [inline Slim/Formats/Movie.pm:277]: forgot to remove lines 277-292 that are only needed for the "process" step

File: Slim/Player/Protocols/HTTP.pm [hunk 2]
```diff
@@ -286,9 +339,59 @@ sub canDirectStream {
...
+	
+	# can't go direct if we are synced or proxy is set by user
+	my $direct = $class->canDirectStream( $client, $song->streamUrl(), $class->getFormatForURL() );
+	return 0 unless $direct;
+	
+	# no header or stripHeader flag has precedence
+	return $direct if $song->stripHeader || !defined $song->track->initial_block_type;
+	
+	# with dynamic header 2, always go direct otherwise only when not seeking
+	if ($song->track->initial_block_type == Slim::Schema::RemoteTrack::INITIAL_BLOCK_ALWAYS || $song->seekdata || $song->track->audio_initiate) {
```
Comment (philippe44) [inline Slim/Player/Protocols/HTTP.pm:357]: Same here, we should not try to test audio_initiate which is not supposed to exist

File: Slim/Player/Protocols/HTTP.pm [hunk 3]
```diff
@@ -286,9 +339,59 @@ sub canDirectStream {
 	return $url;
 }
 
+# TODO: what happens if this method is overloaded in a sub-class that does not call its parents method
+sub canDirectStreamSong {
+	my ( $class, $client, $song ) = @_;
+	
+	# can't go direct if we are synced or proxy is set by user
+	my $direct = $class->canDirectStream( $client, $song->streamUrl(), $class->getFormatForURL() );
+	return 0 unless $direct;
```
Comment (michaelherger) [inline Slim/Player/Protocols/HTTP.pm:351]: Could `$direct` hereafter ever be anything else than "truthy"? Is anything expecting this to be anything other than 0 or 1? If so, please leave a note. If not, please get rid of it and just `return 1` below instead.

Comment (philippe44) [reply Slim/Player/Protocols/HTTP.pm:351]: But I wanted to stop testing here if the canDirectStream tells me it cannot be direct. No need to test further and especially the tests below should not be run if the canDirectStream already said "no". I could have added a if block, but I thought it was more simple to just exit here and have less indentations

**Comment by michaelherger** [comment]
This PR can be closed, right?

**Comment by philippe44** [comment]
yes

**Pull request closed without merging**